### PR TITLE
Upgrade to Realm Sync v10.0.0-beta.10

### PR DESCRIPTION
**Pull request by kneth**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,24 @@
...
+=============================================================
+NOTE: Support for syncing with realm.cloud.io and/or Realm Object Server has been replaced with support for syncing with MongoDB Realm Cloud.
+
+NOTE: This version bumps the Realm file format to version 11. It is not possible to downgrade to earlier versions. Older files will automatically be upgraded to the new file format. Files created by Realm JavaScript prior to v1.0.0, might not be upgradeable. Only [Realm Studio 10.0.0](https://github.com/realm/realm-studio/releases/tag/v10.0.0-beta.1) or later will be able to open the new file format.
+
+### Enhancements
+* None.
+
+### Fixed
+* Fixed validation of idempotent `AddColumn` instruction. This could lead to crashes with the error message `Failed to parse, or apply received changeset: ERROR: AddColumn 'class_X.%3' which already exists`. ([HELP#18016](https://jira.mongodb.org/browse/HELP-18016))
```
Comment (steffenagger) [inline CHANGELOG.md:11]: `'class_X.%3'` sounds like `.%3` might be a specific column-target for a specific db?
I might be wrong, but if not, maybe change it to something more generic e.g. `'class.column'`.

**Review by steffenagger — APPROVED** [review]
See comment.

File: CHANGELOG.md (hunk 1 above)
Comment (kneth) [reply CHANGELOG.md:11]: I'll update the error message. Yes, I believe it is a non-public Jira ticket, and it should be removed.

**Comment by kneth** [comment]
Tests passing locally on MacOS.

**Pull request merged**